tick/survival/convolutional_sccs.py

```diff
@@ -0,0 +1,922 @@
+                              record_every, verbose, seed):
+        # seed cannot be None in SVRG
+        solver_obj = SVRG(step=step, max_iter=max_iter, tol=tol,
+                          print_every=print_every,
+                          record_every=record_every,
+                          verbose=verbose, seed=seed)
+
+        return solver_obj
+
+    def _construct_generator_obj(self, strength_tv_range,
+                                 strength_group_l1_range,
+                                 logspace=True):
+        generators = []
+        if len(strength_tv_range) == 2:
+            if logspace:
+                generators.append(Log10UniformGenerator(*strength_tv_range))
+            else:
+                generators.append(uniform(strength_tv_range))
+        else:
+            generators.append(null_generator)
+
+        if len(strength_group_l1_range) == 2:
+            if logspace:
+                generators.append(
+                    Log10UniformGenerator(*strength_group_l1_range))
+            else:
+                generators.append(uniform(strength_group_l1_range))
+        else:
+            generators.append(null_generator)
+
+        return generators
+
+    # Properties #
+    @property
+    def step(self):
+        return self._step_size
+
+    @step.setter
+    def step(self, value):
+        self._set('_step_size', value)
+        self._solver_obj.step = value
+
+    @property
+    def _strengths(self):
+        return Strengths(self._strength_tv,
+                         self._strength_group_l1)
+
+    @_strengths.setter
+    def _strengths(self, value):
+        if len(value) == 2:
+            self.strength_tv = value[0]
+            self.strength_group_l1 = value[1]
+        else:
+            raise ValueError('strength should be a tuple of length 2.')
+
+    @property
+    def strength_tv(self):
+        return self._strength_tv
+
+    @strength_tv.setter
+    def strength_tv(self, value):
+        if value is None or isinstance(value, float) and value > 0:
+            self._set('_strength_tv', value)
+        else:
+            raise ValueError(
+                'strength_tv should be a float greater than zero.')
+
+    @property
+    def strength_group_l1(self):
+        return self._strength_tv
+
+    @strength_group_l1.setter
+    def strength_group_l1(self, value):
+        if value is None or isinstance(value, float) and value > 0:
+            self._set('_strength_group_l1', value)
+        else:
+            raise ValueError('strength_group_l1 should be a float greater '
+                             'than zero.')
+
+    @property
+    def random_state(self):
+        return self._random_state
+
+    @random_state.setter
+    def random_state(self, value):
+        np.random.seed(value)
+        self._set('_random_state', value)
+
+    @property
+    def n_lags(self):
+        return self._n_lags
+
+    @n_lags.setter
+    def n_lags(self, value):
+        offsets = [0]
```

Comment:
We must check here that it's >= 0 and <= n_intervals - 1

Reply:
Ok for `>= 0`, but we only get to know n_intervals during `_prefit`. This model does not fit very well with sklearn fit / predict framework

Reply:
I finally put the check on `n_lags <= n_intervals - 1` in `_prefit` rather than the setter